pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/PulsarAdmin.java

```diff
@@ -189,7 +191,11 @@ public PulsarAdmin(String serviceUrl,
         this.client = clientBuilder.build();
 
         this.serviceUrl = serviceUrl;
-        root = client.target(serviceUrl);
+        ServiceURI serviceUri = ServiceURI.create(serviceUrl);
+        root = client.target(String.format("%s://%s"
+                , serviceUri.getServiceScheme()
+                , serviceUri.getServiceHosts()[ThreadLocalRandom.current()
+                        .nextInt(serviceUri.getServiceHosts().length)]));
 
```

Comment:
I have a little doubt. Is this fixed the problem that the pulsar function does not support multi-service URLs?

Reply:
> I have a little doubt. Is this fixed the problem that the pulsar function does not support multi-service URLs?

Yes, when we create a Function, the Client initiates a request. Since multi-service URLs are not supported here, a `NumberFormatException` will be thrown.
When you restore this code, then update the jar to the lib, create a Function with cli, you can reproduce the previous bug.